store/rootmulti/store_test.go

```diff
@@ -506,17 +508,15 @@ func getExpectedCommitID(store *Store, ver int64) types.CommitID {
 	}
 }
 
+// hashStores mirrors commitInfo.Hash and commitInfo.toMap but with a different input type
 func hashStores(stores map[types.StoreKey]types.CommitKVStore) []byte {
-	m := make(map[string][]byte, len(stores))
-	for key, store := range stores {
-		name := key.Name()
-		m[name] = storeInfo{
-			Name: name,
-			Core: storeCore{
-				CommitID: store.LastCommitID(),
-				// StoreType: store.GetStoreType(),
-			},
-		}.GetHash()
+	if len(stores) == 0 {
+		return nil
+	}
+	storeMap := make(map[string][]byte, len(stores))
+	for k, v := range stores {
+		storeMap[k.Name()] = v.LastCommitID().Hash
 	}
-	return SimpleHashFromMap(m)
+	rootHash, _, _ := merkle.SimpleProofsFromMap(storeMap)
```

Comment:
this will not be present in tendermint in 0.34, this is the reason it was added here. You will have to keep it in the sdk

Reply:
Hmm... is adding it back not an option? If not, then we need to add the map back that will just set the kv pairs in a deterministic order.

The crucial difference is that `set` should NOT be hashing the values before setting:
https://github.com/cosmos/cosmos-sdk/pull/6069/files#diff-d6645841bf549250d884388c71767a95R35

Then once we set kvpairs in the map in the SDK, we can just sort and call `SimpleHashFromByteSlices` or `SimpleProofsFromByteSlices` instead of the respective `...FromMap` functions.

cc: @ethanfrey @alexanderbez

Reply:
I can double-check with the team. We are doing a push to reduce the code that isn't used in Tendermint and is only kept there for use in the sdk. 

BTW we also removed KV.Pair so you will have to add this to the sdk as well.

Reply:
If being removed from tendermint, I would just move the entire crypto/merkle package to the cosmos sdk. Including all the proof OP stuff.

Or is that staying in tendermint?

I do believe these simple proofs are used in tendermint

Reply:
this was the only one that was not being used in tendermint. The rest have use in tendermint so we opted to keep them